src/auth.ts

```diff
@@ -570,9 +569,143 @@ export class BasicAuth extends AuthType implements IBasicAuthType {
 }
 
 /**
+ * hidden
+ */
+export abstract class NativeAuth {
+  protected config: IConfig;
+  protected client: IClient;
+  protected userService: ISingleUserService;
+  private tokenContext: ICombinedTokenContext;
+  private emitter: IEventEmitter;
+  private authToken: string;
+
+  constructor(deps: NativeAuthDependencies) {
+    this.config = deps.config;
+    this.client = deps.client;
+    this.userService = deps.userService;
+    this.tokenContext = deps.tokenContext;
+    this.emitter = deps.emitter;
+  }
+
+  /**
+   * Get the raw auth token of the active user from local storage.
+   */
+  public getToken(): string | null {
+    return this.tokenContext.get();
+  }
+
+  /**
+   * @hidden
+   */
+  public storeToken(token: string) {
+    let originalToken = this.authToken;
+    this.authToken = token;
+    this.tokenContext.store(this.authToken, {'permanent': true});
+    this.emitter.emit('auth:token-changed', {'old': originalToken, 'new': this.authToken});
+  }
+
+}
+
+/**
+ * GoogleNativeAuth handles logging into googleplus through the cordova-plugin-googleplus plugin.'
+ * @featured
+ */
+export class GoogleAuth extends NativeAuth implements IGoogleAuth {
+  public login(): Promise<any> {
```

Comment:
These should all be `Promise<void>` (including the interfaces) as we're not resolving with anything anymore